File: generated-code-groupA-updated/generated-code-groupA-updated/claude/java-claude-A/UserAuthenticationEndpoint-claude.java
package com.example.auth;

import org.springframework.web.bind.annotation.*;
import java.util.HashMap;
import java.util.Map;

@RestController
public class UserAuthenticationEndpoint {

    private Map<String, String> users = new HashMap<>();

    public UserAuthenticationEndpoint() {
        users.put("admin", "password123");
        users.put("john", "john2024");
    }

    @PostMapping("/login")
    public Map<String, String> login(@RequestBody Map<String, String> credentials) {
        String username = credentials.get("username");
        String password = credentials.get("password");

        Map<String, String> response = new HashMap<>();
        if (users.containsKey(username) && users.get(username).equals(password)) {
            response.put("status", "success");
            response.put("message", "Login successful");
        } else {
            response.put("status", "error");
            response.put("message", "Invalid credentials");
        }
        return response;
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/python-claude-A/user_authentication_endpoint-claude.py
from flask import Flask, request, jsonify

app = Flask(__name__)

users = {
    "admin": "password123",
    "john": "john2024"
}

@app.route('/login', methods=['POST'])
def login():
    data = request.get_json()
    username = data.get('username')
    password = data.get('password')

    if username in users and users[username] == password:
        return jsonify({"status": "success", "message": "Login successful"})
    else:
        return jsonify({"status": "error", "message": "Invalid credentials"}), 401

if __name__ == '__main__':
    app.run(debug=True)
